modules/standard/IO.chpl

```diff
@@ -1715,16 +1776,22 @@ to create a channel to actually perform I/O operations
              whether or not to create the file if it doesn't exist.
              See :type:`iomode`.
 :arg hints: optional argument to specify any hints to the I/O system about
-            this file. See :type:`iohints`.
+            this file. See :record:`ioHintSet`.
 :returns: an open file to the requested resource.
 
 :throws SystemError: Thrown if the file could not be opened.
 */
-proc open(path:string, mode:iomode, hints:iohints=IOHINT_NONE): file throws {
+proc open(path:string, mode:iomode, hints=ioHintSet.empty): file throws {
   return openHelper(path, mode, hints);
 }
 
-private proc openHelper(path:string, mode:iomode, hints:iohints=IOHINT_NONE,
+pragma "last resort"
+deprecated "The 'iohints' type is deprecated; please use a variant of 'open' that takes an 'ioHintSet' instead."
+proc open(path:string, mode:iomode, hints:iohints=IOHINT_NONE): file throws {
+  return openHelper(path, mode, new ioHintSet(hints));
```

Comment:
I think the deprecated version should call the new version of the function rather than performing the same calls it does.  That way if something changes with `openHelper`, we only need to update one place to keep the deprecated and new versions in line with each other.